### PR TITLE
applications: nrf5340_audio: Moved contin_array

### PR DESCRIPTION
**Pull request by koffes**
Tests and code moved

Signed-off-by: Kristoffer Rist Skøien <kristoffer.skoien@nordicsemi.no>

### CODE REVIEW

**Comment by greg-fer** [comment]
@koffes , added a PR that adds missing docs that points at your branch, but I need info from you: https://github.com/koffes/sdk-nrf/pull/2
Can you hold off with merging this PR of yours until my PR is done and merged?

**Review by greg-fer — COMMENTED** [review]
The following docs need to be updated:

- Configuring section on the library documentation page
- Changelog with an entry about the new library

**Review by alexsven — APPROVED** [review]

**Review by rick1082 — APPROVED** [review]

File: include/contin_array.h [hunk 1]
```diff
@@ -9,20 +9,30 @@
...
  * played back in a loop using the results in pcm_cont.
  *
- * @return 0 on success.        -EPERM if any sizes are zero
- *				-ENXIO on NULL pointer.
+ * @retval 0		If the operation was successful.
+ * @retval -EPERM 	If any sizes are zero.
+ * @retval -ENXIO	On NULL pointer.
  */
 int contin_array_create(void *pcm_cont, uint32_t pcm_cont_size, void const *const pcm_finite,
 			uint32_t pcm_finite_size, uint32_t *const finite_pos);
```
Comment (greg-fer) [inline include/contin_array.h:38]: Please add the following lines before the closing #endif to solve the doc build issue:
```
/**
 * @}
 */
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/others/contin_array.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+The user may then call the function to populate the destination array with the pointer ``pcm_cont`` with ``pcm_cont_size`` bytes of data.
+This allows creating ``N`` repetitions of ``pcm_finite`` into ``pcm_cont``, with data fetched in smaller pieces into RAM.
+
+The function keeps track of the current position in ``pcm_finite``, so that the function can be called multiple times and maintain the correct position in ``pcm_finite``.
+Wrapping is handled automatically.
+
+Configuration
+*************
+
+To enable the library, set the :kconfig:option:`CONFIG_CONTIN_ARRAY` Kconfig option to ``y`` in the project configuration file :file:`prj.conf`.
```
Comment (greg-fer) [inline doc/nrf/libraries/others/contin_array.rst:27]: @koffes , do we still need update to this section or the Kconfig was aligned?

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/others/contin_array.rst (hunk 2 above)
Comment (koffes) [reply doc/nrf/libraries/others/contin_array.rst:27]: Awaiting response from reviewers on this issue.

**Review by desowin — APPROVED** [review]

File: include/contin_array.h [hunk 3]
```diff
@@ -9,22 +9,35 @@
...
- *				-ENXIO on NULL pointer.
+ * @retval 0		If the operation was successful.
+ * @retval -EPERM	If any sizes are zero.
+ * @retval -ENXIO	On NULL pointer.
  */
 int contin_array_create(void *pcm_cont, uint32_t pcm_cont_size, void const *const pcm_finite,
 			uint32_t pcm_finite_size, uint32_t *const finite_pos);
 
+/**
+ * @}
```
Comment (erikrobstad) [inline include/contin_array.h:41]: Remove if "@defgroup" is removed?

File: doc/nrf/libraries/others/contin_array.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
+.. _lib_contin_array:
+
+Continuous array
+################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The continuous array library introduces an array that you can loop over, for example if you want to create a period of the Pulse-code modulated (PCM) sine wave.
```
Comment (erikrobstad) [inline doc/nrf/libraries/others/contin_array.rst:10]:
```suggestion
The continuous array library introduces an array that you can loop over, for example if you want to create a period of a Pulse-code modulated (PCM) sine wave.
```

File: include/contin_array.h [hunk 5]
```diff
@@ -9,22 +9,35 @@
 
 #include <zephyr/kernel.h>
 
-/** @brief. Creates a continuous array from a finite array
- * @param pcm_cont		Pointer to the destination array
+/**
+ * @file contin_array.h
+ *
+ * @defgroup contin_array Continuous array
```
[the remark this answers is not in the document]
Comment (greg-fer) [reply include/contin_array.h:15]: This is needed for doxygen API docs. Do not remove this.

File: include/contin_array.h (hunk 3 above)
Comment (greg-fer) [reply include/contin_array.h:41]: This is needed for doxygen API docs. Do not remove this.

File: doc/nrf/libraries/others/contin_array.rst (hunk 2 above)
Comment (greg-fer) [reply doc/nrf/libraries/others/contin_array.rst:19]: I understand that this part is final, @koffes ?

**Review by greg-fer — APPROVED** [review]
Approving, but please squash my commit and let me know about the Configuration part.

**Pull request merged**